src/Init/Data/Ord.lean

```diff
@@ -304,6 +304,27 @@ theorem then_eq_eq {o₁ o₂ : Ordering} : o₁.then o₂ = eq ↔ o₁ = eq 
 theorem then_eq_gt {o₁ o₂ : Ordering} : o₁.then o₂ = gt ↔ o₁ = gt ∨ o₁ = eq ∧ o₂ = gt := by
   cases o₁ <;> cases o₂ <;> decide
 
+@[simp]
+theorem then_lt {o : Ordering} : lt.then o = lt := rfl
```

Comment:
I would argue that for the purposes of the naming scheme, `then` is an infix operator (similar to `Option.bind`, and names like `Std.TreeMap.minKey?_bind_getKey?`), so I would call this lemma `lt_then`.